graphql/directive.go

```diff
@@ -0,0 +1,51 @@
+package graphql
+
+
+import (
+	"reflect"
+)
+
+const (
+	SKIP    = "skip"
+	INCLUDE = "include"
+	IF      = "if"
+)
+
+// findDirectiveWithName checks if any of the directives on a field have the sepcified name (eg skip or include)
+func findDirectiveWithName(directives []*Directive, name string) *Directive {
+	for _, directive := range directives {
+		if directive.Name == name {
+			return directive
+		}
+	}
+	return nil
+}
+
+// shouldIncludeNode validates and checks the value of a skip or include directive
+func shouldIncludeNode(directives []*Directive) (bool, error) {
+	parseIf := func(d *Directive) (bool, error) {
```

Comment:
consider putting this outside the function if you're not using something in scope